colossalai/inference/struct.py

```diff
@@ -167,9 +170,15 @@ class BatchInfo:
     Information to be passed and used for a batch of sequences.
     """
 
+    max_batch_size: int
+    kv_max_split_num: int
+    num_heads: int
+    head_dim: int
     sequences_set: OrderedSet[Sequence] = None
     is_prompts: bool = True
     device: torch.device = None
+    dtype: torch.dtype = None
+    fd_inter_tensor: FDIntermTensors = FDIntermTensors()
 
     def __post_init__(self):
```

Comment:
Is post init here necessary

Reply:
Adding the __post_init__ can ensure the safety of the code and avoid unnecessary bugs.

Reply:
When I delete the __post_init__ and set the default value, it will cause the program to fail. These two ways of writing should not be equivalent.